spotbugs/src/main/java/edu/umd/cs/findbugs/util/MutableClasses.java

```diff
@@ -147,20 +156,55 @@ private boolean looksLikeASetter(Method method) {
             return false;
         }
 
-        // If the method throws an UnsupportedOperationException then we ignore it.
-        ExceptionTable exceptions = method.getExceptionTable();
-        if (exceptions != null) {
-            if (Arrays.asList(exceptions.getExceptionNames()).contains("java.lang.UnsupportedOperationException")) {
-                return false;
+        private String getSig() {
+            String local = sig;
```

Comment:
Rather than the possible `null` assignment followed by double assignment on one line, how about doing a `null` check on `sig`, populating it, then assigning `local = sig` afterward?

Actually, why does `local` exist? We're populating `sig` anyway...

Reply:
Hmm, something is wrong then in GitHub, because you commented an earlier change here. (Not even by me, but by @rovarga, already merged.)